src/Services/PluginInfoService.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Findologic\Services;
+
+use Plenty\Modules\Plugin\Contracts\PluginRepositoryContract;
+use Plenty\Modules\Plugin\Models\Plugin;
+use Plenty\Modules\Plugin\PluginSet\Contracts\PluginSetRepositoryContract;
+use Plenty\Plugin\CachingRepository;
+
+class PluginInfoService
+{
+    const PLUGIN_VERSION_CACHE_KEY_PREFIX = 'plugin_version_';
+
+    /**
+     * @var PluginRepositoryContract|null
+     */
+    private $pluginRepository;
+
+    /**
+     * @var CachingRepository
+     */
+    private $cache;
```

Comment:
If this service is registered as singleton, you should be able to get these via constructor injection out-of-the-box.